### PR TITLE
Enable drone for testing

### LINKED ISSUE #3: Add tests for various s3 products
- [ ] minio
- [ ] ceph
- [ ] scality

### PR DESCRIPTION
**Pull request by patrickjahns**
Fixes #3 

Migrated test suite from core repository and enabled it to be run in drone

### CODE REVIEW

File: tests/Amazons3Test.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Files_External\Tests\Storage;
+
+use OCA\Files_External\Lib\Storage\AmazonS3;
```
Comment (DeepDiver1975) [inline tests/Amazons3Test.php]: @patrickjahns this is the wrong class - should be \OCA\FilesExternalS3/Lib/Storage/AmazonS3

File: tests/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
+require_once __DIR__ . '/../vendor/autoload.php';
+define('PHPUNIT_RUN', 1);
```
Comment (DeepDiver1975) [inline tests/bootstrap.php]: can be removed

File: tests/bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+require_once __DIR__ . '/../../../lib/base.php';
+
+\OC::$composerAutoloader->addPsr4('Test\\', OC::$SERVERROOT . '/tests/lib/', true);
+\OC::$composerAutoloader->addPsr4('Tests\\', OC::$SERVERROOT . '/tests/', true);
+
+if (!class_exists('PHPUnit_Framework_TestCase')) {
```
Comment (DeepDiver1975) [inline tests/bootstrap.php]: can be removed

File: lib/Storage/AmazonS3.php [hunk 4]
```diff
@@ -404,7 +404,7 @@ public function touch($path, $mtime = null) {
 			$mtime = time();
 		}
 		$metadata = [
-			'lastmodified' => gmdate(\Aws\Common\Enum\DateFormat::RFC1123, $mtime)
+			'lastmodified' => gmdate(\DateTime::RFC1123, $mtime)
```
Comment (DeepDiver1975) [inline lib/Storage/AmazonS3.php]: please rebase on top of https://github.com/owncloud/files_external_s3/pull/2/commits

**Comment by patrickjahns** [comment]
@DeepDiver1975 
Configured and running now - test failures are due to changes in deleteobjects call - ref: https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-s3-2006-03-01.html#deleteobjects

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975
> Configured and running now - test failures are due to changes in deleteobjects call - ref: https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-s3-2006-03-01.html#deleteobjects

I'll have a look tomorrow - THX for you support

**Comment by DeepDiver1975** [comment]
the remaining failing tests show the miss behavior as described in https://github.com/owncloud/core/issues/29619

**Comment by DeepDiver1975** [comment]
I will merge this - issues need to be fixed in follow up prs

**Pull request merged**